src/platform_win.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#if defined(_MSC_VER)
+#   define CATA_COMPILER_MSVC _MSC_VER
```

Comment:
Quoting myself from the previous PR:
> I don't find the aliases for the platform defines to be an improvement, they just trade a standard define for a nonstandard one most of the time, and a pair of standard defines for a nonstandard one in a few places.

Just bundle up the windows-specific includes, don't alias the platform defines.
As a result of that, you're including this header in 17 files where it is adding negligible benefit, polluting those files with the (gigantic) windows headers.